tests/testsuite.src/run_misc.at

```diff
@@ -15251,3 +15251,324 @@ AT_CHECK([$COMPILE prog.cob])
 AT_CHECK([$COBCRUN_DIRECT ./prog], [0], [1.2345E-5 1.2345E-5], [])
 
 AT_CLEANUP
+
+
+AT_SETUP([profiling and codegen])
+AT_KEYWORDS([EVALUATE])
+
+AT_DATA([prog.cob], [
+IDENTIFICATION DIVISION.
+PROGRAM-ID. PROG.
+
+DATA DIVISION.
+WORKING-STORAGE SECTION.
+    01 STR PIC X(6) VALUE "..-..-".
+    01 OFFSET BINARY-LONG UNSIGNED.
+
+PROCEDURE DIVISION.
+    MOVE 1 TO OFFSET
+    PERFORM UNTIL OFFSET > LENGTH OF STR
+        CALL "TestRecurse" USING STR OFFSET
+    END-PERFORM.
+
+END PROGRAM PROG.
+
+
+IDENTIFICATION DIVISION.
+PROGRAM-ID. TestRecurse IS RECURSIVE.
+
+DATA DIVISION.
+LINKAGE SECTION.
+    01 LK-STR PIC X ANY LENGTH.
+    01 LK-OFFSET BINARY-LONG UNSIGNED.
+
+PROCEDURE DIVISION USING LK-STR LK-OFFSET.
+    DISPLAY "LK-OFFSET: " LK-OFFSET
+    EVALUATE LK-STR(LK-OFFSET:1)
+        WHEN "."
+            ADD 1 TO LK-OFFSET
+            CALL "TestRecurse" USING LK-STR LK-OFFSET
+        WHEN OTHER
+            ADD 1 TO LK-OFFSET
+    END-EVALUATE.
+
+END PROGRAM TestRecurse.
+])
+
+AT_CHECK([$COMPILE --free prog.cob])
+
+AT_CHECK([$COBCRUN_DIRECT ./prog], [0],
+[LK-OFFSET: 0000000001
+LK-OFFSET: 0000000002
+LK-OFFSET: 0000000003
+LK-OFFSET: 0000000004
+LK-OFFSET: 0000000005
+LK-OFFSET: 0000000006
+])
+
+AT_CHECK([$COMPILE --free -fprof prog.cob])
+
+AT_CHECK([$COBCRUN_DIRECT ./prog], [0],
+[LK-OFFSET: 0000000001
+LK-OFFSET: 0000000002
+LK-OFFSET: 0000000003
+LK-OFFSET: 0000000004
+LK-OFFSET: 0000000005
+LK-OFFSET: 0000000006
+])
+AT_DATA([prog.cob], [
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. PROG.
+
+       DATA DIVISION.
+       WORKING-STORAGE SECTION.
+           01 STR PIC X(6) VALUE "..-..-".
+           01 OFFSET BINARY-LONG UNSIGNED.
+
+       PROCEDURE DIVISION.
+           MOVE 1 TO OFFSET
+           PERFORM UNTIL OFFSET > LENGTH OF STR
+               CALL "TestRecurse" USING STR OFFSET
+           END-PERFORM.
+
+       END PROGRAM PROG.
+
+
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. TestRecurse IS RECURSIVE.
+
+       DATA DIVISION.
+       LINKAGE SECTION.
+           01 LK-STR.
+              03 LK-CHR PIC X OCCURS 6 INDEXED BY LK-IDX.
+           01 LK-OFFSET BINARY-LONG UNSIGNED.
+
+       PROCEDURE DIVISION USING LK-STR LK-OFFSET.
+           DISPLAY "LK-OFFSET: " LK-OFFSET
+           SET LK-IDX TO LK-OFFSET
+           SEARCH LK-CHR
+               AT END
+                   IF LK-OFFSET > 20
+                      GOBACK
+                   END-IF
+                   ADD 1 TO LK-OFFSET
+                   CALL "TestRecurse" USING LK-STR LK-OFFSET
+               WHEN LK-CHR (LK-IDX) = "."
+                   ADD 1 TO LK-OFFSET
+           END-SEARCH.
+       END PROGRAM TestRecurse.
+])
+
+AT_CHECK([$COMPILE --free prog.cob])
```

Comment:
no need for `--free` here (and if, then we should use the "new" variant with free as option), same below